stock_dynamic_routing/models/stock_move.py

```diff
@@ -238,13 +239,27 @@ def _apply_routing_rule_pull(self, routing_details):
         """
         pickings_to_check_for_emptiness = self.env["stock.picking"]
         move_ids_to_assign_per_location = defaultdict(list)
-        for move in self:
-            routing_rule = routing_details[move].rule
-            if not routing_rule.method == "pull":
+        move_ids_to_assign_nonrelocated = []
+        next_moves_to_update = self.browse()
+        routing_to_apply = [
+            (move, detail.rule) for move, detail in routing_details.items()
+        ]
+        for move, routing_rule in routing_to_apply:
+            if routing_rule.method == "push":
+                # In this case, we should not assign the move inside the
+                # pull dynamic routing. The push move must first be added before the
+                # initial move is assigned. Otherwise, when the destination
+                # location of the initial move is changed by the push rule, the
+                # putaway won't be recomputed as it is already assigned.
+                continue
+
+            if not routing_rule:
```

Comment:
it makes more sense to but this condition before routing_rule.method == push.

Reply:
done (amended the current commit)